### PR TITLE
Enable deduplication of constants with NaNs. (#3328)

### PR DESCRIPTION
**Pull request by chadrosier**
Summary:
Constants that contain NaN will not be de-duplicated because isEqual always returns false for this case.  Instead, perform a bitwise comparison to achieve the expected behavior.

### CODE REVIEW

File: tests/unittests/GraphOptzTest.cpp [hunk 1]
```diff
@@ -2371,6 +2371,58 @@ TEST_F(GraphOptz, VarsCSE) {
...
 }
 
+TEST_F(GraphOptz, VarsCSENaN) {
+  // Create two variables that are Private, are not trainable, have no writers
+  // and include NaNs. The first two variables have the same data, and so should
+  // be combined via variable CSE.  In particular, the NaN constants should not
+  // prevent the variables from being combine.
+  auto *input1 = mod_.createConstant(ElemKind::FloatTy, {5}, "input1");
+  auto *input2 = mod_.createConstant(ElemKind::FloatTy, {5}, "input2");
+  auto nan = std::numeric_limits<float>::quiet_NaN();
```
Comment (rdzhabarov) [inline tests/unittests/GraphOptzTest.cpp]: interesting, is it any different from using NAN value?

File: tests/unittests/GraphOptzTest.cpp [hunk 2]
```diff
@@ -2371,6 +2371,58 @@ TEST_F(GraphOptz, VarsCSE) {
...
+  input1->getHandle() = {nan, nan, nan, nan, nan};
+  input2->getHandle() = {nan, nan, nan, nan, nan};
```
Comment (rdzhabarov) [inline tests/unittests/GraphOptzTest.cpp]: I'd add some variety NAN + non NAN values in the constants to make sure it's handled properly.

**Comment by rdzhabarov** [comment]
@chadrosier make sure to fix the test failures

File: include/glow/Base/Tensor.h [hunk 3]
```diff
@@ -629,6 +643,16 @@ class Tensor final {
     }
     return numExceedingError == 0;
   }
+
+  bool isBitwiseEqualImpl(const Tensor &other) const {
+    auto const *myData = getRawDataPointer<uint8_t>();
```
Comment (rdzhabarov) [inline include/glow/Base/Tensor.h]: I think you need: `getUnsafePtr`

File: tests/unittests/GraphOptzTest.cpp (hunk 1 above)
Comment (chadrosier) [reply tests/unittests/GraphOptzTest.cpp]: This was an arbitrary choice.  I'll update the pull request to use the NAN value.

File: include/glow/Base/Tensor.h (hunk 3 above)
Comment (chadrosier) [reply include/glow/Base/Tensor.h]: Correct.  Thanks, Roman.  I missed this because I was testing on a release build (without asserts).

**Comment by chadrosier** [comment]
Unfortunately, I'm not used to this development flow and my attempt to rebase my fork/branch to ToT went *way* off course.

Updated pull request here: https://github.com/pytorch/glow/pull/3376
This should address all of your comments, Roman.

**Pull request closed without merging**